### ISSUE TITLE
Improper use of ga() to provide options

### ISSUE THREAD

**Issue by DevLab2425** [react-ga/react-ga#116]
By using the [Google Analytics Chrome Extension](https://chrome.google.com/webstore/detail/google-analytics-debugger/jnkmfdileelhofjcijamephohjechhna), it was revealed that the provided options were being set improperly (see screenshot of extension output below).

Ideally, the options are set using `ga('set', ....);` for each option provided.

<img width="1044" alt="screen shot 2017-03-31 at 10 57 23 am" src="https://cloud.githubusercontent.com/assets/8385375/24556488/88bbc3d8-1602-11e7-9c68-747ba801e143.png">

**Comment by DevLab2425** [comment]
PR submitted:
https://github.com/react-ga/react-ga/pull/117

**Comment by paul-barry-kenzan** [comment]
@yangshun 
Any movement on this issue and PR?

**Comment by SimeonC** [comment]
It looks like you are mixing up the usages of `create` and `set`. What you pass through to `ReactGA.initialize` is the fields from the "Create Only Fields" here; https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference.
You should be using `ReactGA.set(...)` for those fields as it states in the warning. `set` Reference https://github.com/react-ga/react-ga#reactgasetfieldsobject.

The PR you suggested would prevent the setting of Create Only Fields as you pass them all through the set GA action.

I think you just need to change your code to:
```
ReactGA.initialize('UA000000-1');
ReactGA.set({
  appName: '...',
  appId: '...',
  appVersion: '...',
  hostname: '...'
});
```

**Comment by DevLab2425** [comment]
Thanks for the feedback @SimeonC 

The README states that `gaOptions` can be passed through the `initialize()` method (https://github.com/react-ga/react-ga#reactgainitializegatrackingid-options), but that results in the error described above. 

I can certainly change my app to utilize the `.set()` method. However, this does not resolve the incorrect usage that's documented in React-GA's README. PR #117 can be merged to resolve the error and allow users to continue to pass options via `initialize()`, or the library should be updated to disallow passing options through via `initialize()`, and encourage the usage you suggested.

Let me know how I can help.

**Comment by SimeonC** [comment]
Good point, I think the best option is to update the README which I've done in a separate PR.
You should be able to pass through options in initialize, it's just that it has to be clear that you can only send through the Create Only Fields in initialize.

**Comment by SimeonC** [comment]
Closed via #134
